src/ffi.rs

```diff
@@ -66,20 +46,20 @@ pub unsafe extern "C" fn newton_distribute_bodies(
     };
 
     let bodies = distributor.distribution();
-    let field = &mut *field;
-    field.bodies = bodies;
+    let environment = &mut *environment;
+    environment.bodies = bodies;
 }
 
 #[no_mangle]
-pub unsafe extern "C" fn newton_step(field: *mut Field) {
-    let field = &mut *field;
-    field.update()
+pub unsafe extern "C" fn newton_step(environment: *mut Environment) {
+    let environment = &mut *environment;
+    environment.update()
 }
 
 #[no_mangle]
-pub unsafe extern "C" fn newton_body_pos(field: *const Field, id: u32) -> NewtonPoint {
-    let field = &*field;
-    match field.bodies.get(&id) {
+pub unsafe extern "C" fn newton_body_pos(environment: *const Environment, id: u32) -> NewtonPoint {
```

Comment:
why do u use `*const` here ?

Reply:
`*const` means an immutable reference. Here we receive the reference to environment, which we can not change.